cargo-spatial/src/local.rs

```diff
@@ -0,0 +1,53 @@
+use crate::config::Config;
+use crate::opt::*;
+use std::ffi::OsString;
+use std::path::*;
+use std::process;
+use tap::*;
+
+/// Prepares and launches a local deployment.
+///
+/// Before launching the deployment, this will first run code generation and build
+/// workers in the project. Assumes that the current working directory is the root
+/// directory of the project, i.e. the directory that has the `Spatial.toml` file.
+pub fn launch(_opt: &Opt, _local: &Local, launch: &LocalLaunch) {
+    let config = Config::load().expect("Failed to load configuration");
+
+    // Run codegen and such.
+    crate::codegen::run_codegen(&config).expect("Failed to run codegen");
```

Comment:
Does this swallow the errors from `codegen`? It would be preferable to bubble those up

Reply:
It doesn't swallow the errors, rather it will panic if there's an error in code generation, which isn't the best way to report errors to the user. I'll tweak it to pass the error along in a more idiomatic manner, though I think error handling/reporting in general could use another pass (it's the part that I was least sure how to handle well).